src/main/java/org/jenkinsci/plugins/ewm/facets/WorkspaceBrowserFacet.java

```diff
@@ -0,0 +1,54 @@
+package org.jenkinsci.plugins.ewm.facets;
+
+import hudson.model.Fingerprint;
+import jenkins.model.FingerprintFacet;
+import org.jenkinsci.plugins.ewm.model.ExternalWorkspace;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * TODO JAVADOC
+ *
+ * @author Alexandru Somai
+ */
+public class WorkspaceBrowserFacet extends FingerprintFacet {
+
+    public WorkspaceBrowserFacet(@Nonnull Fingerprint fingerprint, long timestamp) {
+        super(fingerprint, timestamp);
+    }
+
+    /**
+     * TODO JAVADOC
+     *
+     * @return
+     */
+    @Restricted(NoExternalUse.class)
+    @SuppressWarnings("unused")
+    public String getWorkspaceLink() {
+        ExternalWorkspace workspace = getWorkspace();
+        if (workspace == null) {
+            return "#";
+        }
+
+        Fingerprint.BuildPtr original = getFingerprint().getOriginal();
+        if (original == null) {
+            return "#";
+        }
+
+        return original.getJob().getAbsoluteUrl() + original.getRun().getSearchUrl() +
```

Comment:
You do not actually need to hold workspaces anywhere else, because starting from this change you will have the Fingerprint database with it's automatic removal features :)

So Root action can actually go to Fingerprint by ID and retrieve Workspace information for browsing from it